hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -1008,9 +1008,11 @@ private RegionLocations locateRegionInMeta(TableName tableName, byte[] row, bool
       }
       // Query the meta region
       long pauseBase = connectionConfig.getPauseMillis();
-      takeUserRegionLock();
       final long lockStartTime = EnvironmentEdgeManager.currentTime();
+      boolean lockedUserRegion = false;
       try {
+        takeUserRegionLock();
```

Comment:
I don't see much difference in the overall behaviour by just bring the lock acquisition inside the try. It lock acquiring fails, we are still halting the flow, and not really doing anything extra in the finally block.

Reply:
No. Now if locking taking fails, the exception will be catched at line 1099, and there is pause of sleeping at line 1132 before next retry. Without this PR, the exception when taking lock will just throw to the outer layer.

Reply:
Thanks. Haven't seen that catch block before.